Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -211,6 +211,62 @@ bool TextureCache::TCacheEntryBase::OverlapsMemoryRange(u32 range_address, u32 r
 	return true;
 }
 
+void TextureCache::TCacheEntryBase::do_partial_texture_updates()
+{
+	const bool isPaletteTexture = (format== GX_TF_C4 || format == GX_TF_C8 || format == GX_TF_C14X2 || format >= 0x10000);
+
+	// As long as there's no upscaling support, there's no point in doing this when doing efb to ram
+	// TODO: Only update normal textures. Would there be any point in updating efb copies?
+	// TODO: Are texture updates for paletted textures possible?
+	// Efb copies might be scaled, while normal textures are not. So don't try to update normal textures with scaled efb copies
+	if (!g_ActiveConfig.bSkipEFBCopyToRam || IsEfbCopy() || isPaletteTexture
+		|| (g_ActiveConfig.bCopyEFBScaled && g_ActiveConfig.iEFBScale != SCALE_1X))
+		return;
+
+	const u32 block_width = TexDecoder_GetBlockWidthInTexels(format);
+	const u32 block_height = TexDecoder_GetBlockHeightInTexels(format);
+
+	u32 expanded_width = (native_width + block_width - 1) & ~(block_width - 1);
+	//u32 expanded_height = (native_height + block_height - 1) & ~(block_height - 1);
+
+	u32 numBlocksX = expanded_width / block_width;
+	//u32 numBlocksY = expanded_height / block_height;
+
+	u32 cacheLinesPerRow;
+	// TODO: Is this check correct?
+	if (format == GX_TF_RGBA8) // RGBA takes two cache lines per block; all others take one
+		cacheLinesPerRow = numBlocksX * 2;
+	else
+		cacheLinesPerRow = numBlocksX;
+
+	TexCache::iterator iter = textures_by_address.lower_bound(addr);
+	TexCache::iterator iterend = textures_by_address.upper_bound(addr + size_in_bytes);
```

Comment:
This doesn't find textures which starts below addr. As we're only able to search by the base address of the textures, we likely have to start at addr - max_texture_size.
IIRC 1024*1024*4 is the max texture size.